src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.opendistroforelasticsearch.knn.plugin.script;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+
+public class KNNScoringUtil {
+    private static Logger logger = LogManager.getLogger(KNNScoringUtil.class);
+
+    public static float l2Squared(float[] queryVector, float[] inputVector) {
+        float squaredDistance = 0;
+        for (int i = 0; i < inputVector.length; i++) {
+            float diff = queryVector[i]-inputVector[i];
+            squaredDistance += diff * diff;
+        }
+        return squaredDistance;
+    }
+
+    public static float cosinesimilOptimized(float[] queryVector, float[] inputVector, float normQueryVector) {
+        float dotProduct = 0.0f;
+        float normInputVector = 0.0f;
+        for (int i = 0; i < queryVector.length; i++) {
+            dotProduct += queryVector[i] * inputVector[i];
+            normInputVector += inputVector[i] * inputVector[i];
+        }
+        float normalizedProduct = normQueryVector * normInputVector;
+        try {
+            return (float) (dotProduct / (Math.sqrt(normalizedProduct)));
+        } catch(ArithmeticException ex) {
```

Comment:
How about we use if/else to make code cleaner like the method below line47?

Reply:
Java will throw `ArithmeticException` for integer division by zero, but not throw `ArithmeticException` for float division by zero, check this [stack overflow question](https://stackoverflow.com/questions/12954193/why-does-division-by-zero-with-floating-point-or-double-precision-numbers-not)

Reply:
Good catch. Able to catch this in unit tests. Added ```if``` condition to check for division by zero issue. Adding ```if``` check could have minor performance impact. Will keep an eye in performance tests

Reply:
Thanks, looks good to me.